src/components/Pagination/Pagination.tsx

```diff
@@ -2,0 +1,74 @@
+import React from 'react';
+
+interface PaginationProps {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: (page: number) => void;
+}
+
+export const Pagination: React.FC<PaginationProps> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const totalPages = Math.ceil(total / perPage);
+
+  const handlePageChange = (page: number) => {
+    let newPage = page;
+
+    if (newPage < 1) {
+      newPage = 1;
+    } else if (newPage > totalPages) {
+      newPage = totalPages;
+    }
+
+    onPageChange(newPage);
+  };
```

Comment:
Code quality issue: The `handlePageChange` function could be simplified. Instead of manually setting the value of `newPage` to 1 or `totalPages` when it's out of those bounds, you can use the `Math.max` and `Math.min` functions to ensure the value is within the correct range. This will make your code shorter and easier to understand.